presto-main/src/main/java/com/facebook/presto/sql/planner/SymbolReferenceExtractor.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.facebook.presto.sql.planner;
+
+import com.facebook.presto.sql.tree.DefaultTraversalVisitor;
+import com.facebook.presto.sql.tree.Expression;
+import com.facebook.presto.sql.tree.SymbolReference;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class SymbolReferenceExtractor
```

Comment:
There shouldn't be subqueries anymore in this stage of planning. For lambdas though you should extract symbol references which are captures. I still stand that `SymbolReferenceExtractor` should be removed and you should use `DependencyExtractor#extractUnique(com.facebook.presto.sql.tree.Expression)` instead. However, `DependencyExtractor.SymbolBuilderVisitor` should be fixed to return capture symbol references for lambdas.